src/main/java/be/xplore/pricescraper/utils/ProductMatcher.java

```diff
@@ -0,0 +1,74 @@
+package be.xplore.pricescraper.utils;
+
+import org.apache.commons.text.similarity.LevenshteinDistance;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ProductMatcher {
+
+    private final int MATCH_THRESHOLD;
+
+    public ProductMatcher(@Value("${productmatcher.ingredients.threshold}") int threshold) {
+        this.MATCH_THRESHOLD = threshold;
+    }
+
+    /**
+     * Match 2 products by passing their ingredient strings. Shops often desribe product names slightly different,
+     * by checking the ingredient list we can verify that two products are the same as these should be described
+     * accurately by the vendors.
+     *
+     * @return boolean is same product
+     */
+    public boolean isSameProductByIngredients(String ingredients1, String ingredients2) {
+        return matchProductsByIngredients(ingredients1, ingredients2) < MATCH_THRESHOLD;
+    }
+
+    /**
+     * Match 2 products by passing their ingredient strings. Shops often desribe product names slightly different,
+     * by checking the ingredient list we can verify that two products are the same as these should be described
+     * accurately by the vendors.
+     *
+     * @return degree matched (lower is better)
+     */
+    public int matchProductsByIngredients(String ingredients1, String ingredients2) {
+        ingredients1 = normalizeIngredientsString(ingredients1);
+        ingredients2 = normalizeIngredientsString(ingredients2);
+        LevenshteinDistance ls = LevenshteinDistance.getDefaultInstance();
```

Comment:
could be worth specifying LevenshteinDistance as a static so you only construct one